modules/core/src/main/java/org/locationtech/jts/geom/util/GeometryExtracter.java

```diff
@@ -123,29 +123,29 @@ public static List extract(Geometry geom, int ordinal)
    */
   public GeometryExtracter(Class clz, List comps)
   {
-  	this.ordinal = toOrdinal(clz);
+  	this.geometryType = toGeometryType(clz);
     this.comps = comps;
   }
   
   /**
    * Constructs a filter with a list in which to store the elements found.
    * 
-   * @param ordinal Geometry ordinal constant as defined by for example {@link Geometry#ORDINAL_POINT} (null means all types)
+   * @param ordinal Geometry ordinal constant as defined by for example {@link Geometry#TYPECODE_POINT} (null means all types)
    * @param comps the list to extract into
    */
-  public GeometryExtracter(int ordinal, List comps)
+  public GeometryExtracter(String geometryType, List comps)
   {
-  	this.ordinal = ordinal;
+  	this.geometryType = geometryType;
     this.comps = comps;
   }
   
-  private static boolean isLinear(Geometry geom, int ordinal) {
-	return geom.getOrdinal() == Geometry.ORDINAL_LINEARRING && ordinal == Geometry.ORDINAL_LINESTRING;
+  private static boolean isLinear(Geometry geom, String geometryType) {
+	return geom.getGeometryType() == Geometry.TYPENAME_LINEARRING && geometryType == Geometry.TYPENAME_LINESTRING;
   }
 
   public void filter(Geometry geom)
   {
-	if (ordinal == -1 || geom.getOrdinal() == ordinal || isLinear(geom, ordinal))
+	if (geometryType == null || geom.getGeometryType() == geometryType || isLinear(geom, geometryType))
```

Comment:
Not sure about the presence of isLinear here.  Wouldn't that mean that linear geometries are always added no matter what the requested type?

Reply:
The original logic extracted LinearRings when the clz was set to LineString, which I didn't expect at first but I suppose that's how isAssignableFrom works. So that's why I added this exception to the rule. But perhaps it's the name isLinear which isn't right?

Reply:
Ah yes, much clearer - taken. :)